### PR TITLE
Node.js documentation

### PR DESCRIPTION
**Pull request by gj**
PR checklist:

- [x] TypeScript API docs
- [x] Node.js version of Quickstart
- [x] Node.js version of Application Types
- [x] Working with Node.js Types
- [x] Node.js version of RBAC
- [x] Node.js version of ABAC
- [x] Node.js version of Context
- [ ] Node.js version of Multiple Actor Types
- [ ] Node.js version of Resources with Inheritance
- [x] Node.js version of REPL docs
- [x] Grep docs to ensure "Node.js" is added anywhere we currently enumerate supported languages.
- [x] Automate build of TypeScript API docs
- [x] Link built TypeScript API docs on Libraries page
- [x] Added changelog entry.
- [x] Install instructions
- [x] JavaScript Library Page

### CODE REVIEW

File: docs/examples/abac/nodejs/01-simple.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+const { Oso } = require('oso');
+
+const oso = new Oso();
+
+const EXPENSES = [
+  { amount: 500, submitted_by: 'alice', location: 'NYC', project_id: 2 },
```
Comment (leina05) [inline docs/examples/abac/nodejs/01-simple.js:6]: Is it weird to use camel case in JavaScript?

Comment (gj) [reply docs/examples/abac/nodejs/01-simple.js:6]: camelCase is idiomatic; snake_case is not

File: docs/using/libraries/node/index.rst [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+your policy (see :ref:`application-types`).
+
+Class Instances
+^^^^^^^^^^^^^^^
+
+Any ``new``-able JavaScript object (including ES6-style classes) can be
+constructed from inside an oso policy using the :ref:`operator-new` operator if
+the constructor (a ``class`` or ``function`` that responds to `the new operator
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new>`_)
+has been **registered** using the ``#registerClass`` method. An example of
```
Comment (leina05) [inline docs/using/libraries/node/index.rst:42]: Is the `#` a JS convention also, or is this left over from ruby?

Comment (gj) [reply docs/using/libraries/node/index.rst:42]: it's not a JS convention; we could drop it

File: docs/examples/abac/nodejs/01-simple.js (hunk 1 above)
Comment (leina05) [reply docs/examples/abac/nodejs/01-simple.js:6]: Okay, there are a few other places where snake case is used in the JS examples, but I assume that's just for consistency with the rest of the docs for now. We need to go through all the docs for this issue anyway, can be a follow-up.

File: docs/using/libraries/node/index.rst (hunk 2 above)
Comment (leina05) [reply docs/using/libraries/node/index.rst:42]: It's not used in other places to refer to JS methods (that I saw), so maybe best to drop it.

File: docs/examples/abac/nodejs/01-simple.js (hunk 1 above)
Comment (gj) [reply docs/examples/abac/nodejs/01-simple.js:6]: > Okay, there are a few other places where snake case is used in the JS examples, but I assume that's just for consistency with the rest of the docs for now. We need to go through all the docs for this issue anyway, can be a follow-up.

assumption is correct; agree on the follow-up

File: docs/using/libraries/node/index.rst (hunk 2 above)
Comment (gj) [reply docs/using/libraries/node/index.rst:42]: agree; follow-up

**Pull request merged**